### PR TITLE
Filter all terms in bulk surface fluxes

### LINKED ISSUE #698: Possible bug in bulk drag formulation when using filtered veloctites
Hi,

The behavior of the scalar and momentum fluxes is different when using filtered velocities. For momentum (Jᵘ = - Cᴰ |U| ρu), the wind speed is filtered, but the ρu term isn't. For scalar fluxes (J = -ρ₀ Cᵀ |U| Δϕ), both the wind speed and scalar difference are filtered. 

**A run using (20 minute) filtering:**
Image

**A run with no filtering:**
Image

https://github.com/NumericalEarth/Breeze.jl/blob/674ddcecd6c4781a8476363ed8b099d907c9d16c/src/BoundaryConditions/bulk_drag.jl#L72-L81

https://github.com/NumericalEarth/Breeze.jl/blob/674ddcecd6c4781a8476363ed8b099d907c9d16c/src/BoundaryConditions/bulk_scalar_fluxes.jl#L123-L138

It's not totally clear to me what's intended from https://link.springer.com/article/10.1007/s10546-025-00934-9?utm_source=researchgate.net&utm_medium=article, my intuition says all the fields should be filtered..

@glwagner @navidcy

### PR DESCRIPTION
**Pull request by glwagner**
Closes #698.

When a `FilteredSurfaceVelocities` is attached to a bulk boundary condition,
every field that enters the bulk formula is now read from a filtered source:
wind speed `|U|`, the velocity (or scalar) difference, and the virtual
potential temperature `θᵥ` used inside the stability-dependent transfer
coefficient. Previously only `|U|` was filtered while `ρu`, `Δϕ`-readers,
and `θᵥ` in `bulk_coefficient` were read instantaneously — that's the
inconsistency reported in the issue.

## Momentum drag formula

Restructured from `Jᵘ = -Cᴰ |U| ρu` to the scalar-flux-parallel form

  Jᵘ = -ρ₀ Cᴰ |U| u

because Monin–Obukhov similarity is a profile law for `u`, not `ρu`
([Shin, Yang & Howland 2025](https://link.springer.com/article/10.1007/s10546-025-00934-9)).
`ρ₀` is the surface density from `(p₀, T₀, constants)`, exactly as in
`BulkSensibleHeatFluxFunction`/`BulkVaporFluxFunction`. When `BulkDrag` is
constructed without an explicit `surface_temperature` (allowed for constant
`coefficient`), `materialize_bulk_drag` fills T₀ from the reference-state
potential temperature so `ρ₀` is always well-defined.

## Where the filtered θᵥ lives

The user-facing API doesn't change: `FilteredSurfaceVelocities` is still
constructed once and shared across drag + sensible + vapor BCs. Internally
it grows a `θᵥ :: Field{Center, Center, Nothing}` and an independent
`last_θᵥ_update` Ref so multiple BCs sharing the same `fv` only trigger
the exponential filter once per stage.

## Files

| File | What changed |
|------|--------------|
| `src/BoundaryConditions/filtered_surface_state.jl` | `θᵥ` field + Ref, constructor/Adapt updates, `update_θᵥ!`/`initialize_θᵥ!`, dedup-aware helpers |
| `src/BoundaryConditions/polynomial_bulk_coefficient.jl` | `stability_corrected_coefficient` dispatches on filtered θᵥ source |
| `src/BoundaryConditions/bulk_drag.jl` | New `surface_pressure`/`thermodynamic_constants` fields, new `Jᵘ = -ρ₀ Cᴰ Ũ u` formula |
| `src/BoundaryConditions/BoundaryConditions.jl` | `surface_velocity_at_face` helpers; thread `surface_pressure`/`constants` through `materialize_bulk_drag`; reference-state T₀ fallback |
| `src/BoundaryConditions/update_boundary_conditions.jl` | Drive filtered θᵥ update/init from each bulk BC's `PolynomialCoefficient` |
| `test/forcing_and_boundary_conditions.jl` | New "BulkDrag uses ρ₀, filtered u and θᵥ" testset |
| `test/polynomial_bulk_coefficients.jl` | θᵥ field/Ref tests, `update_θᵥ!` unit test, existing drag-flux test updated to new formula and T₀ default |

## Verified locally

- `polynomial_bulk_coefficients`: 192/192 pass
- `forcing_and_boundary_conditions` "Bulk boundary conditions" testset: 28/28 pass (other testsets unchanged and green)
- End-to-end smoke test of the `prescribed_sea_surface_temperature` example (reduced grid): runs `time_step!` cleanly, filtered ū integrates from 0 toward the set value, filtered θ̄ᵥ updates, and τˣ has the expected sign.

cc @navidcy @mmr0

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/polynomial_bulk_coefficients.jl [hunk 1]
```diff
@@ -816,16 +840,17 @@ using GPUArraysCore: @allowscalar
...
-        expected_filtered = -Cᴰ * Ũ² * ρu_sfc / U_f
-
-        # What the flux WOULD be with instantaneous velocity (u₀ = 5):
-        Ũ²_inst = u₀^2 + FT(gustiness)^2
-        expected_unfiltered = -Cᴰ * Ũ²_inst * ρu_sfc / u₀
+        # New formula: τ = -ρ₀ * Cᴰ * Ũ * u, with `u` and `Ũ` read from filtered fields.
+        p₀ = model_surface_pressure(model.dynamics)
+        T₀_default = Breeze.AtmosphereModels.default_drag_surface_temperature(model.dynamics,
+                                                                                grid,
+                                                                                model.thermodynamic_constants)
```
Comment (navidcy) [inline test/polynomial_bulk_coefficients.jl]:
```suggestion
        T₀_default = Breeze.AtmosphereModels.default_drag_surface_temperature(model.dynamics,
                                                                              grid,
                                                                              model.thermodynamic_constants)
```

File: test/polynomial_bulk_coefficients.jl [hunk 2]
```diff
@@ -816,16 +840,17 @@ using GPUArraysCore: @allowscalar
...
+        ρ₀ = surface_density(p₀, T₀_default, model.thermodynamic_constants)
+        Ũ_filtered   = sqrt(U_f^2 + FT(gustiness)^2)
+        expected_filtered   = - ρ₀ * Cᴰ * Ũ_filtered * U_f
```
Comment (navidcy) [inline test/polynomial_bulk_coefficients.jl]:
```suggestion
        Ũ_filtered = sqrt(U_f^2 + FT(gustiness)^2)
        expected_filtered = - ρ₀ * Cᴰ * Ũ_filtered * U_f
```

File: test/polynomial_bulk_coefficients.jl (hunk 1 above)
Comment (navidcy) [inline test/polynomial_bulk_coefficients.jl]:
```suggestion
        T₀_default = Breeze.AtmosphereModels.default_drag_surface_temperature(model.dynamics,
                                                                              grid,
                                                                              model.thermodynamic_constants)
```

File: test/polynomial_bulk_coefficients.jl [hunk 3]
```diff
@@ -792,7 +814,9 @@ using GPUArraysCore: @allowscalar
 
         fv = FilteredSurfaceVelocities(grid; filter_timescale=τ_filter)
 
-        # Build model with constant coefficient and filtered velocities
+        # Build model with constant coefficient and filtered velocities. No
+        # `surface_temperature` is supplied; `materialize_bulk_drag` fills it from
+        # the reference-state surface temperature so ρ₀ can be evaluated.
         ρu_bcs = FieldBoundaryConditions(bottom=Breeze.BulkDrag(coefficient=Cᴰ, gustiness=gustiness,
                                                                  filtered_velocities=fv))
```
Comment (navidcy) [inline test/polynomial_bulk_coefficients.jl]:
```suggestion
ρu_bcs = FieldBoundaryConditions(bottom=Breeze.BulkDrag(coefficient=Cᴰ, gustiness=gustiness, filtered_velocities=fv))
```

File: test/polynomial_bulk_coefficients.jl [hunk 4]
```diff
@@ -397,6 +399,24 @@ using GPUArraysCore: @allowscalar
...
+        set!(θᵥ_source, 305.0)
+
+        Breeze.BoundaryConditions.update_θᵥ!(fv, θᵥ_source, grid, 2.0)
+        ε = 2.0 / 20.0
+        expected = (0.0 + ε * 305.0) / (1 + ε)
+        @test fv.θᵥ[1, 1, 1] ≈ expected atol=1e-10
+
+        # Initialize sets the field directly from the source (no time integration)
+        Breeze.BoundaryConditions.initialize_θᵥ!(fv, θᵥ_source, grid)
+        @test fv.θᵥ[1, 1, 1] ≈ 305.0 atol=1e-10
```
Comment (navidcy) [inline test/polynomial_bulk_coefficients.jl]:
```suggestion
        θᵢ = 305
        θᵥ_source = CenterField(grid)
        set!(θᵥ_source, θᵢ)

        Breeze.BoundaryConditions.update_θᵥ!(fv, θᵥ_source, grid, 2.0)
        ε = 2.0 / 20.0
        expected = (0.0 + ε * θᵢ) / (1 + ε)
        @test fv.θᵥ[1, 1, 1] ≈ expected atol=1e-10

        # Initialize sets the field directly from the source (no time integration)
        Breeze.BoundaryConditions.initialize_θᵥ!(fv, θᵥ_source, grid)
        @test fv.θᵥ[1, 1, 1] ≈ θᵢ atol=1e-10
```

File: test/forcing_and_boundary_conditions.jl [hunk 5]
```diff
@@ -129,6 +129,16 @@ end
...
+
+        # CompressibleDynamics has no default surface temperature for BulkDrag;
+        # constructing a model without an explicit surface_temperature must error.
+        compressible_dyn = CompressibleDynamics(SplitExplicitTimeDiscretization(substeps=2);
+                                                reference_potential_temperature = FT(300),
+                                                surface_pressure = FT(1e5),
+                                                standard_pressure = FT(1e5))
+        ρu_bcs_no_T₀ = FieldBoundaryConditions(bottom=BulkDrag(coefficient=Cᴰ, gustiness=gustiness))
+        @test_throws ArgumentError AtmosphereModel(grid; dynamics=compressible_dyn,
+                                                          boundary_conditions=(; ρu=ρu_bcs_no_T₀))
```
Comment (navidcy) [inline test/forcing_and_boundary_conditions.jl]:
```suggestion
        @test_throws ArgumentError AtmosphereModel(grid; dynamics=compressible_dyn,
                                                         boundary_conditions=(; ρu=ρu_bcs_no_T₀))
```

File: src/BoundaryConditions/bulk_drag.jl [hunk 6]
```diff
@@ -32,17 +61,20 @@ where `Cᴰ` is the drag coefficient, `|U| = √(u² + v² + gustiness²)` is th
...
 - `gustiness`: Minimum wind speed to prevent singularities when winds are calm (default: `0`)
-- `surface_temperature`: Surface temperature, required when using `PolynomialCoefficient`
-  with stability correction. Can be a `Field`, `Function`, or `Number`. (default: `nothing`)
+- `surface_temperature`: Surface temperature, used to compute `ρ₀` and required when
+  using `PolynomialCoefficient` with stability correction. Can be a `Field`,
+  `Function`, or `Number`. (default: `nothing`)
 - `filtered_velocities`: A [`FilteredSurfaceVelocities`](@ref) for temporally filtered
-  wind speed in the bulk formula. If `nothing` (default), instantaneous velocity is used.
+  wind speed, near-surface velocity, and `θᵥ` in the bulk formula. If `nothing`
+  (default), instantaneous fields are used.
```
Comment (navidcy) [inline src/BoundaryConditions/bulk_drag.jl:74]: add the relevant ref(s) in the docstring?

File: src/AnelasticEquations/AnelasticEquations.jl [hunk 7]
```diff
@@ -32,7 +32,8 @@ using Oceananigans.Operators: Δzᵃᵃᶜ, Δzᵃᵃᶠ, divᶜᶜᶜ, Δzᶜ
 using Oceananigans.Solvers: Solvers, solve!, FourierTridiagonalPoissonSolver, AbstractHomogeneousNeumannFormulation
 using Oceananigans.Utils: prettysummary, launch!
 
-using Breeze.Thermodynamics: ReferenceState, MoistureMassFractions, mixture_gas_constant
+using Breeze.Thermodynamics: ReferenceState, MoistureMassFractions, mixture_gas_constant,
+                              dry_air_gas_constant
```
Comment (navidcy) [inline src/AnelasticEquations/AnelasticEquations.jl]:
```suggestion
using Breeze.Thermodynamics: ReferenceState, MoistureMassFractions, mixture_gas_constant,
                             dry_air_gas_constant
```

**Comment by navidcy** [comment]
@mmr0 if you re-run the figs shown in #698 would be useful.
I can review the code changes, no need for you to do that

**Comment by mmr0** [comment]
> @mmr0 if you re-run the figs shown in https://github.com/NumericalEarth/Breeze.jl/issues/698 would be useful.

@navidcy I (finally)  re-ran using this PR and the fields look good to me!

jacquet_fig2b_xy_surface_fluxes

**Review by mmr0 — APPROVED** [review]
I've not carefully reviewed this I just ran my model script with the PR and compared the surface fluxes using filtered velocities to those not using filtered velocities
